### ISSUE TITLE
torchvision.io.read_image should take file path or image encoded as bytes

### ISSUE THREAD

**Issue by agunapal** [pytorch/vision#7751]
### 🚀 The feature

Currently `torchvision.io.read_image()` expects a file path as input.

In many practical cases, the image is encoded as `bytes` 

It would be nice to have a torchvision method to read this and convert to a tensor

Expected behavior
```
img_t = torchvision.io.read_image(str or bytes)
```

### Motivation, pitch

Ease of use
Good user experience

### Alternatives

I use the following code

```
image = Image.open(io.BytesIO(image))
image = transforms.ToTensor()(image)
```

**Comment by NicolasHug** [comment]
I think you're looking for  `decode_image()`? https://pytorch.org/vision/main/generated/torchvision.io.decode_image.html#torchvision.io.decode_image

**Comment by agunapal** [comment]
@NicolasHug This doesn't work because of the following
```
RuntimeError: image::decode_image() Expected a value of type 'Tensor' for argument '_0' but instead found type 'bytes'.

```

**Comment by NicolasHug** [comment]
yes you'll need to convert the Bytes object to a tensor - probably with https://pytorch.org/docs/stable/generated/torch.frombuffer.html ?
